doc/api/worker.md

```diff
@@ -0,0 +1,493 @@
+`MessagePort`s on that `MessageChannel` to the other thread through a
+pre-existing channel, such as the global one.
+
+See [`port.postMessage()`][] for more information on how messages are passed,
+and what kind of JavaScript values can be successfully transported through
+the thread barrier.
+
+For example:
+
+```js
+const assert = require('assert');
+const { Worker, MessageChannel, MessagePort, isMainThread } = require('worker');
+if (isMainThread) {
+  const worker = new Worker(__filename);
+  const subChannel = new MessageChannel();
+  worker.postMessage({ hereIsYourPort: subChannel.port1 }, [subChannel.port1]);
+  subChannel.port2.on('message', (value) => {
+    console.log('received:', value);
+  });
+} else {
+  require('worker').once('workerMessage', (value) => {
+    assert(value.hereIsYourPort instanceof MessagePort);
+    value.hereIsYourPort.postMessage('the worker is sending this');
+    value.hereIsYourPort.close();
+  });
+}
+```
+
+### new Worker(filename, options)
+
+* `filename` {string} The absolute path to the Worker’s main script.
+  If `options.eval` is true, this is a string containing JavaScript code rather
+  than a path.
+* `options` {Object}
+  * `eval` {boolean} If true, interpret the first argument to the constructor
+    as a script that is executed once the worker is online.
+  * `data` {any} Any JavaScript value that will be cloned and made
+    available as [`require('worker').workerData`][]. The cloning will occur as
+    described in the [HTML structured clone algorithm][], and an error will be
+    thrown if the object cannot be cloned (e.g. because it contains
+    `function`s).
+  * stdin {boolean} If this is set to `true`, then `worker.stdin` will
+    provide a writable stream whose contents will appear as `process.stdin`
+    inside the Worker. By default, no data is provided.
+  * stdout {boolean} If this is set to `true`, then `worker.stdout` will
+    not automatically be piped through to `process.stdout` in the parent.
+  * stderr {boolean} If this is set to `true`, then `worker.stderr` will
+    not automatically be piped through to `process.stderr` in the parent.
+
+### Event: 'error'
+<!-- YAML
+added: REPLACEME
+-->
+
+* `err` {Error}
+
+The `'error'` event is emitted if the worker thread throws an uncaught
+exception. In that case, the worker will be terminated.
+
+### Event: 'exit'
+<!-- YAML
+added: REPLACEME
+-->
+
+* `exitCode` {integer}
+
+The `'exit'` event is emitted once the worker has stopped. If the worker
+exited by calling [`process.exit()`][], the `exitCode` parameter will be the
+passed exit code. If the worker was terminated, the `exitCode` parameter will
+be `1`.
+
+### Event: 'message'
+<!-- YAML
+added: REPLACEME
+-->
+
+* `value` {any} The transmitted value
+
+The `'message'` event is emitted when the worker thread has invoked
+[`require('worker').postMessage()`][]. See the [`port.on('message')`][] event
+for more details.
+
+### Event: 'online'
```

Comment:
Just a comment: `ready` is the intuitive name for this event for me.

Reply:
I’m totally fine with switching, just gonna wait a short time to wait for more people to react to your comment – this is inspired by the `online` event of `cluster` workers, but I agree that `ready` is *probably* better.